File: 2026/April/LC-2075-Decode-the-Slanted-Ciphertext/solution.cpp
// Problem Link: https://leetcode.com/problems/decode-the-slanted-ciphertext/

class Solution {
    public:
        string decodeCiphertext(string encodedText, int rows) {
            bool isEmpty = true;
            for (char c : encodedText) {
                if (c != ' ') isEmpty = false;
            }
            if (isEmpty) return "";
            int cols = encodedText.length() / rows;
            string res = "";
            for (int col = 0; col < cols; col++) {
                int r = 0, c = col;
                while (r < rows && c < cols) {
                    res += encodedText[r * cols + c];
                    r++;
                    c++;
                }
            }
            int k = res.length() - 1;
            while (res[k] == ' ') {
                k--;
            }
            return res.substr(0, k + 1);
        }
    };

File: 2026/April/LC-2075-Decode-the-Slanted-Ciphertext/solution.py
# Problem Link: https://leetcode.com/problems/decode-the-slanted-ciphertext/


class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        res = ""
        cols = len(encodedText) // rows
        for col in range(cols):
            r, c = 0, col
            while r < rows and c < cols:
                res += encodedText[r*cols + c]
                r += 1
                c += 1
        return res.rstrip()